pkg/virt-operator/kubevirt.go

```diff
@@ -668,7 +668,13 @@ func (c *KubeVirtController) execute(key string) error {
 	}
 
 	if !reflect.DeepEqual(kv.Finalizers, kvCopy.Finalizers) {
-		if _, err := c.clientset.KubeVirt(kvCopy.ObjectMeta.Namespace).Update(kvCopy); err != nil {
+		kvNew, err := c.clientset.KubeVirt(kvCopy.ObjectMeta.Namespace).Get(kvCopy.Name, &metav1.GetOptions{})
+		if err != nil {
+			logger.Reason(err).Errorf("Could not get the KubeVirt resource.")
+			return err
+		}
+		kvNew.Finalizers = kvCopy.Finalizers
+		if _, err := c.clientset.KubeVirt(kvNew.ObjectMeta.Namespace).Update(kvNew); err != nil {
```

Comment:
As you said yourself, we need a patch here. Could you use it instead?

Reply:
Ok but the proper way to do that is by rewriting this function, and probably others, to build a patch instead of modifying a copy of the CR. That will take a while, and this is blocking single-replica testing. Would you consider merging this band-aid while I work on the better version?

Reply:
Considering

> > @jean-edouard if this does not succeed for pretty much forever, do we then hot-loop on e.g. the status update?
> 
> Unsure but that's a good theory

I would want to continue with the root cause analysis. If this is the case it is still a very severe issue and I would give it higher priority than any follow-up testing. What do you think?

Reply:
> Considering
> 
> > > @jean-edouard if this does not succeed for pretty much forever, do we then hot-loop on e.g. the status update?
> > 
> > 
> > Unsure but that's a good theory
> 
> I would want to continue with the root cause analysis. If this is the case it is still a very severe issue and I would give it higher priority than any follow-up testing. What do you think?

Yeah that's fair. I am off today but I will have another look tomorrow. Thanks for the review!